implementations/rust/ockam/ockam_core/src/error/inner/mod.rs

```diff
@@ -0,0 +1,183 @@
+#[allow(unused)]
+use super::code::{Kind, Origin};
+#[cfg(feature = "std")]
+use crate::compat::{borrow::ToOwned, vec::Vec};
+use crate::compat::{boxed::Box, error::Error as ErrorTrait, string::String};
+use crate::error::code::ErrorCode;
+use serde::{Deserialize, Serialize};
+
+mod formatting;
+#[cfg(all(feature = "std", feature = "error-traces"))]
+mod trace_config;
+
+type BoxDynErr = Box<dyn ErrorTrait + Send + Sync + 'static>;
+
+/// The internal error type, which [`super::NewError`] wraps.
+///
+/// When an allocator is available, this is `Box`ed for performance reasons,
+/// (although some of the fields here are a little goofy in low level contexts).
+#[derive(Serialize, Deserialize)]
+pub(super) struct ErrorData {
+    pub(super) code: ErrorCode,
+    // FIXME: `#[cfg()]`ing the fields out changes the serialization format. Not good.
+    #[cfg(feature = "std")]
+    payload: Vec<PayloadEntry>,
```

Comment:
Could `payload` and `source_location` not be made available for `alloc`?

Reply:
Generally I think we want to keep the Error type as light-weight as possible for embedded targets. Memory is tiiiight and we're already using too much in other areas of the codebase! 😅